### ISSUE TITLE
Zone Retrieve Request Fails

### ISSUE THREAD

**Issue by Aliasr** [horrorho/InflatableDonkey#109]
Hi all. 

Looks that Apple recently implemented some changes that makes the zoneRetrieveRequest fail. I haven't been able to download my backups for about two weeks now. Has anyone faced a similar issue recently?

**Comment by lostaconfused** [comment]
Yes, it is completely not working now. Maybe the person that provided the recent fix can provide another, hopefully. @RobLinux

**Comment by RobLinux** [comment]
Not only that changed, but the file record retrieval is broken too now.
Zones can be retrieved through GSA login but then I'm stuck

**Comment by ymohammad** [comment]
@RobLinux what is GSA login. When I am trying using InflatableDonkey, it is failing with an unauthorized error for ZoneRetrieveRequest. How did you achieve it?

**Comment by snak3y** [comment]
@RobLinux Just getting back into all of this now as I see many versions are broken. 

Looks like some of GSA flow changed as well? I tried to go through some of my old GSA code and I can't even get the M2 back when I send M1. Looks like there's a lot I need to catch up on...

**Comment by RobLinux** [comment]
@snak3y it seems that for some accounts the password derivation function is a bit different and can result in a wrong M2 computing. I can show you if you contact me over skype / mail.

**Comment by snak3y** [comment]
@RobLinux Message sent! Hopefully we can get this working.

**Comment by RobLinux** [comment]
answered you!
